internal-dns/src/config.rs

```diff
@@ -378,23 +419,6 @@ impl DnsConfigBuilder {
             (zone.dns_name(), vec![DnsRecord::Aaaa(zone_ip)])
         });
 
-        let scrimlet_srv_records =
-            self.scrimlets.clone().into_iter().map(|(location, addr)| {
-                let srv = DnsRecord::Srv(dns_service_client::types::Srv {
-                    prio: 0,
-                    weight: 0,
-                    port: addr.port(),
-                    target: format!("{location}.scrimlet.{}", DNS_ZONE),
-                });
-                (ServiceName::Scrimlet(location).dns_name(), vec![srv])
-            });
-
-        let scrimlet_aaaa_records =
-            self.scrimlets.into_iter().map(|(location, addr)| {
-                let aaaa = DnsRecord::Aaaa(*addr.ip());
-                (format!("{location}.scrimlet"), vec![aaaa])
-            });
-
```

Comment:
tl;dr: I brought this up in oxide-control-plane chat and sync'd with @rcgoodfellow and @internet-diglett and I believe these DNS names can be removed.

More details:

- I could not find any references to these DNS names anywhere in Omicron.  I looked for "switch0.scrimlet" and "switch1.scrimlet" and even ".scrimlet" (in case it was being filled into a template).  If they were used programmatically, they'd show up as `ServiceName::Scrimlet`.  The only place that's used is [a comment explaining some context](https://github.com/oxidecomputer/omicron/blob/ddd036760d2bef6fdda4c1c18469f253333e4a29/nexus/src/app/sagas/switch_port_settings_common.rs#L504-L517).
- What I got from chat is what that comment says: these DNS names were added at some point after dogfood was set up.  It was then discovered that they couldn't be used for whatever they were going to be used for because these DNS names only get set up at RSS-time, so systems like dogfood were just never going to have them.  Instead, a different mechanism was created to find these zones when needed.
- From what I can tell, the RSS code that assigned these DNS names chose the switch location (switch0 vs. switch1) arbitrarily, based on which one's sled id came lexicographically first.  That's potentially inconsistent with what MGS reported as switch0 and switch1.  So these names were potentially wrong on any systems where they _were_ set up.
- It should be possible to augment the code in this PR to reliably establish correct DNS names for these, provided we have an inventory collection with the information that we need from MGS.  This way, in the future, we _can_ use DNS for this.  I didn't add this to this PR.